Add `random_query` example (#3704)
bevyengine/bevy: examples/README.md

`hierarchy` | [`ecs/hierarchy.rs`](./ecs/hierarchy.rs) | Creates a hierarchy of parents and children entities
`iter_combinations` | [`ecs/iter_combinations.rs`](./ecs/iter_combinations.rs) | Shows how to iterate over combinations of query results.
`parallel_query` | [`ecs/parallel_query.rs`](./ecs/parallel_query.rs) | Illustrates parallel queries with `ParallelIterator`

> `random_query` | [`ecs/random_query.rs`](./ecs/random_query.rs) | Get random components from a query with `rand::seq::IteratorRandom`

Reviewer: ```suggestion
`random_query` | [`ecs/random_query.rs`](./ecs/random_query.rs) | Sample random elements of a query with `rand::seq::IteratorRandom`
```
The returned items are items of the iterator, not individual components

Author: Ah thank you. I went back and forth on how to word that.
